### DISCUSSION TITLE
Исправление синтаксических ошибок в переводе на русский от Reforged Eden 1.11

### DISCUSSION THREAD

**Discussion by SirMaxBlack** [SnakeSel/Empyrion-ru#5] [General]
Привет, заметил, что в игре, взаимодействуя с некоторыми приборами или НПС, окно диалога открывается, но текста никакого нет.
Порылся в файле \Empyrion - Galactic Survival\Content\Scenarios\Reforged Eden Rus\Content\Configuration\Dialogues.csv
Нашел моменты, когда оригинальный текст и теги были перенесены в поле перевод неверно.

В некоторых ситуациях @d0@q0 шел слитно в переводе там, где должен иметь пробел
Также, в игре ломались диалоги из за неверного переноса в перевод следующих символов:       ,"@d0
(в оригинальном тексте нет ковычки, а в переводе она появилась). Если сделать как в оригинальном тексте (без ковычек) то нпс и приборы начинают показывать текст для взаимодействия.

Раньше тут небыло никакого текста. После фикса он появился.
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/b0b7646e-3c75-4fc3-bce6-5aab5970755d)

Вот как выглядит неправильный перенос тега (например, строка 6219
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/f9d737b6-816d-4b85-b9ca-156362d9c6c8)

Пофиксил файл и все что смог найти, нашел.
В игре проверил, вроде пока старых проблем не обнаружил. Если найду еще что, добавлю обновленный файл.

[Dialogues.csv](https://github.com/SnakeSel/Empyrion-ru/files/15051482/Dialogues.csv)

**Comment by SirMaxBlack** [comment]
В процессе игры на базе эдена аналогичную ситуацию нашел с инженером гидропоники. Но пока не нашел способ, как его пофиксить.
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/de4245bb-1914-488d-85db-40327666cd0e)

**Comment by SnakeSel** [comment]
По поводу кавычек. Так экранируется строка, если в ней встречается символ разделителя. В данном случае это запятая (,). Если убрать кавычки, то запятая в тексе посчитается концом поля. Вы уверены, что экранированные строки не работают? Вроде с этим проблем небыло.

Это видно даже на вашем скриншоте, полный текст выглядит так:
`@d0<color=#ff8000>Этот экстрактор в настоящее время не является частью сети.</color>\nВключите режим обновления на вашем <color=#00ffff>{Eden_ControlInfo}</color>, чтобы добавить этот экстрактор в сеть.\ nИли вы можете провести еще один опрос, если вам снова понадобится модернизировать первичный промышленный экстрактор воды.\n<color=#ffffff><i>Примечание для многопользовательской игры: этот экстрактор, возможно, уже был добавлен в сеть другим игроком.</i ></color>`

**Reply by SirMaxBlack** [reply]
К сожалению, я пришел к этому только методом научного тыка. 
Как делал:
Брал половину оригинала(не перевод) и заменял эту же часть в переведенном файле.
Если проблема не уходила, то возвращал обратно и лез в следующий блок перевода.
Повторяя несколько раз, находил тексты, из за которых диалоги в игре были пустые. Причем иногда диалоги пустые вообще в другом месте, не связанные с первоначальным меню.
Далее, находил несколько таких "пустых" окон в разных местах и снова искал в файле.
Далее, все найденные строки сравнил с оригиналом, и там во всех были отличия именно в ковычках. 
Понимаю, что нужно для зеркалирования запятых, чтобы переводы не переходили в другие языковые поля. Но смог только так пофиксить. 

Если есть более правильный вариант, могу сделать корректнее с вашей подскаской.

**Comment by SnakeSel** [comment]
Текст отсутствовал в диалогах на файлах из последнего коммита? Я недавно правил много ошибок в дилогах, но все сводилось к кривым макросам (то что в {}) в переводе.
Меня смущает, что в вашем файле есть другие отличия. например отсутствует перевод строк dialogue_G4FOO0q, dialogue_aumSSSm

**Reply by SirMaxBlack** [reply]
В самом начале, когда только столкнулся с первым пустым диалогом, методом исключения нашел эти две строки и вернул их обратно на оригинал, убрав перевод. Тут мой косяк есть, обратно не вернул.

**Comment by SnakeSel** [comment]
Опять же, я недавно взламывал базы, и диалог работал. а от тоже в кавычках
`"@d0<color=#ff8000>Эта цифра кода </color><color=#ff0000>ниже</color><color=#ff8000>, чем вы предполагаете.</color>"`
Так что кавычки тут не при чем

**Comment by SnakeSel** [comment]
Если диалог пустой, выведите консоль (~), там отобразятся ошибки и можно понять причину

**Reply by SirMaxBlack** [reply]
Есть станция, на которой можно проверить.
Там есть нпс, занимающийся гидропоникой.

![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/9cb9bab9-9f00-43ee-a10c-65447479a80f)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/88297b6e-6bdd-4e7f-be7a-99c3c314c318)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/009b8dae-2412-4965-9d23-0a3ec5831d9e)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/7feaf4e2-5ab4-459c-bcba-100bc3d9289c)

Вот так выглядит окно у него
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/a59f854c-d1ed-45f7-8996-cd2b1681279e)

А так консоль по нему
 (из двух скринов склеил, в одно окно не влезло)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/9c457436-e3eb-4c09-8dba-1b8a28c5e792)

**Reply by SirMaxBlack** [reply]
Вот тут трабла со спутником 

![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/35f9ad57-b0c2-4a84-8d86-115963bb7f3e)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/dea9ca5d-bbbb-4123-9a29-5814bbd1db35)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/f3c0c575-abff-47e0-8e9c-eaabe00449c0)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/82f6a67f-32b0-41ec-8c8b-a0972c5cdd18)

**Reply by SirMaxBlack** [reply]
Тут "Вторичный промышленный экстрактор воды"
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/d5563217-5546-41a9-b62b-8cdc2c2ef648)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/d7e337d4-2737-4a1c-9608-2ef3f133676a)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/d42c64e7-0446-4034-bbfe-5a9bbbda7560)

**Reply by SirMaxBlack** [reply]
Тут "Первичный промышленный экстрактор воды"
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/579dd3d0-769e-4bcc-85ca-e7bb5e28a098)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/5762e099-663f-4276-bc8d-57256e04c394)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/d37863a5-4494-4b98-a86b-f8369b6373c4)

**Comment by SnakeSel** [comment]
Проверьте экстрактор на последнем коммите. Думаю там дело было в `</i >`
Нашел еще пару ошибок, поправил.

У инженера гидропоники и консоли взлома спутника одна ошибка. Вот какая, пока не понял. Там диалог очень большой. (Система грузит сразу все варианты). Вроде ошибка CS1009 это неверный символ после **\\**. Но я ничего такого не вижу.

Можно идти и проверять все пункты диалога в файле Dialogues.ecf, Dialogue Name: Console_Reports_Exploration_Start, но это требует уйму времени.

**Reply by SirMaxBlack** [reply]
Пошел по старому пути, при диагностике спутников (С них у меня все это и началось ковыряние).
Проследил по одной нашей с вами переписке и увидел, что если заменить на оригинал диалог dialogue_G4FOO0q (Оставить его без перевода), то окно взаимодействия с интерфейсом сразу начинает работать корректно.

![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/c800fd25-518c-49af-bf6b-eee106c2ea54)

**Reply by SirMaxBlack** [reply]
Нашел гидропонику.
Если вернуть на оригинал диалог eden_dialogue_GiKOO8i, то инженер начинает работать корректно!)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/dc5f4e81-2eba-46f3-bcb7-246e5c85c2ea)

**Reply by SirMaxBlack** [reply]
Может ли неверный перевод на новую строку так повлиять?


отдельных культур.\n\ nОни не дают 

Если да, то тексте еще есть места с таким
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/03fbae37-c405-48cb-bb99-089d676e5aaf)

**Reply by SnakeSel** [reply]
Вы правы. пробел как раз не должен идти после \\. Исправил в последнем коммите. Диалог на спутнике заработал. Проверьте экстрактор.
Спасибо за помощь!

**Reply by SirMaxBlack** [reply]
Отлично. Все заработало!
И первичный и вторичный экстрактор выдают корректно информацию.
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/b756972a-70df-4863-8079-3dc5515296e0)


Спасибо, что занимаетесь таким благим делом!!
В стиме в комментах написал людям, что появились обновы в переводе)
Подскажите, пожалуйста. Если еще найду пустые окна, мне писать в этой же теме или создать новую?

**Reply by SnakeSel** [reply]
Можно сюда, уведомления приходят, так что не потеряется сообщение

**Reply by SirMaxBlack** [reply]
Играл и в процессе нашел заброшенный корабль The legacy.
Зайдя в него и прыгнув в телепорт, попал на какую-то станцию заброшенную.
В ней нашел терминал, открываю, там аналогичная проблема с пустым окном:
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/4e5323db-ae14-4ebd-8948-5d56babeb689)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/630595bb-7025-435e-a554-e03d5a3c3fd4)

Но, такого "SM_AR_ControlStation_Init" в Dialogues.ecf не нашел
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/8a8d2890-3117-4ae4-bc29-538e716cab86)


Куда нужно копать? 
Может оно самим сценарием поломано, раз такого нет даже в ecf?

**Reply by SnakeSel** [reply]
Это из сюжетки оригинальной игры. Вы файлы для нее заменяли? 
Я проходил сюжетку в оригинале с файлами из этого репозитория, все работало. Но могли и сломать потом

**Reply by SirMaxBlack** [reply]
О, фак. Не знал что нужно еще заменить и основные файлы, думал только RE надо.

Сейчас  скачал последний вариант перевода и заменил файлы, к старой ошибке еще вот это выпало:
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/c24bb55a-773e-4468-b984-b3300d40ef49)




Файлы взял из этой папки и закинул в корень игры, как по инструкции
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/75050f50-75c1-4ffe-a72b-b0d63f65149e)

**Reply by SirMaxBlack** [reply]
Прошу простить.
Не знаю почему, но у меня был не косякнутый ecf файл. 
Хз как так получилось, но файлы отличаются.


Удалил полностью папку и перекачал заново в стиме.
В новом файле есть нужные строки
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/cca92bf0-97a4-4afd-9bf7-859f45884c12)

А в варианте, который качал из комьюнити стима, таких строк нет.
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/3490d714-61d7-4a60-94e9-f34a12114ad7)

Чтобы больше танцев небыло, удалил все изменения комьюнити и оставил только ваш перевод.

Еще раз извиняюсь за лишний мозготреп!(

**Reply by SnakeSel** [reply]
~~Интересно, стал еще подтягивать диалоги из RE.~~ Упс, уже разобрались.
Тогда на будущее:
Чтобы проверить косяк это перевода или нет, выйдите в главное меню и переключитесь на английский язык, потом проверьте диалог. (без перезапуска сохранения изменения языка не подхватятся, поэтому в любом случае выходить до главного меню)

**Reply by SirMaxBlack** [reply]
Переключился на англ, но все равно не находит нужный диалог, хотя он есть в файле ecf
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/85971349-3057-4bd5-9c09-ddfcc9e7bbe1)




А как игра работает?
Она совмещает оба файла ecf (оригинал игры и RE) или только один какой то использует?


Возможно и правда сам сценарий навернулся гдет
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/ad13ea44-06c2-4267-b850-592633ae461c)
![image](https://github.com/SnakeSel/Empyrion-ru/assets/6567430/927ec411-dfd2-478a-908a-9b62bf3742aa)

**Reply by SnakeSel** [reply]
Поэтому All4X4 и говорил, что не стоит править .ecf
Т.к. их изменения надо очень тщательно отслеживать.

У меня только один ItemsConfig.ecf и тот правится скриптом, что позволяет не отслеживать изменения, а обновлять всегда актуальный. И то приходится его постоянно мониторить на обновление файла

**Reply by SirMaxBlack** [reply]
Дополнил прошлый коммент.

**Reply by SnakeSel** [reply]
> Возможно и правда сам сценарий навернулся гдет

По идее этого сценария не должно быть в RE. Это миссия знакомства с легаси в оригинале. В RE для этого есть миссия "поиска пропавших колонистов" (Не помню точное название, 2-я за UCH).
Учитывая что и в английском не работает, тут ничего не поделаешь.

**Reply by SirMaxBlack** [reply]
Попробую связаться с разрабами и указать на траблу. Т.к. еще некотоыре терминалы этого корабля не работают)
![Uploading image.png…]()



Спасибо, что указали, как правильно диагностировать проблему(переключения языков)

**Comment by SnakeSel** [comment]
> А как игра работает?
> Она совмещает оба файла ecf (оригинал игры и RE) или только один какой то использует?

Как я понимаю, так же как с текстом. Если есть в RE, берет оттуда, если нет лезет в оригинал
